plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,182 @@
+package io.jenkins.plugins.casc;
+
+import io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class SchemaGeneration {
+
+
+    final static JSONObject schemaTemplateObject = new JSONObject()
+        .put("$schema", "http://json-schema.org/draft-07/schema#")
+        .put("id", "http://jenkins.io/configuration-as-code#")
+        .put("description", "Jenkins Configuration as Code")
+        .put("type", "object");
+
+    public static JSONObject generateSchema() {
+
+        /**
+         * The initial template for the JSON Schema
+         */
+
+        JSONObject schemaObject = new JSONObject(schemaTemplateObject.toString());
+
+        /**
+         * This generates the schema for the root configurators
+         * Iterates over the root elements and adds them to the schema.
+         */
+
+        JSONObject rootConfiguratorObject = new JSONObject();
```

Comment:
Please split up this large method into smaller methods.

Reply:
Done :)